code/game/machinery/doors/door.dm

```diff
@@ -167,7 +167,7 @@
 		return
 	if(!requiresID())
 		user = null //so allowed(user) always succeeds
-	if(allowed(user) || user.can_advanced_admin_interact())
+	if(allowed(user) || (istype(user) && user.can_advanced_admin_interact()))

```

Comment:
This isn't the best way to fix this; this is happening because the `user` is set to `null` when the door doesn't require an ID.